### PR TITLE
Kati 53 Keep user signed in

### PR DESCRIPTION
**Pull request by toannguyen71195**

### CODE REVIEW

File: app/src/main/java/com/kms/cura/view/activity/LoginActivity.java [hunk 1]
```diff
@@ -188,4 +223,78 @@ protected void onResume() {
...
+    }
+
+    private boolean checkSignIn() {
+        File file = new File(getFilesDir(), SIGNED_IN_INFO);
+        if (file.exists())
+            return true;
+        return false;
+    }
+
+    private void autoSignIn() {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/LoginActivity.java]: This method should be in UserController

File: app/src/main/java/com/kms/cura/view/activity/LoginActivity.java [hunk 2]
```diff
@@ -188,4 +223,78 @@ protected void onResume() {
...
+            String email = scanner.nextLine();
+            String pwd = scanner.nextLine();
+            scanner.close();
+            UserController.userLogin(email, decrypt(pwd));
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void saveLoginInfo() {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/LoginActivity.java]: This method should be in UserController

File: app/src/main/java/com/kms/cura/view/activity/LoginActivity.java [hunk 3]
```diff
@@ -188,4 +223,78 @@ protected void onResume() {
...
+        String p = Base64.encodeToString(bytes, Base64.DEFAULT);
+        return p;
+    }
+
+    private static byte[] base64Decode(String property) throws IOException {
+        byte[] p = Base64.decode(property, Base64.DEFAULT);
+        return p;
+    }
+
+    private boolean checkSignIn() {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/LoginActivity.java]: This method should be in UserController

File: app/src/main/java/com/kms/cura/view/activity/LoginActivity.java [hunk 4]
```diff
@@ -188,4 +223,78 @@ protected void onResume() {
...
+        }
+        return property;
+    }
+
+    private static String base64Encode(byte[] bytes) {
+        String p = Base64.encodeToString(bytes, Base64.DEFAULT);
+        return p;
+    }
+
+    private static byte[] base64Decode(String property) throws IOException {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/LoginActivity.java]: This method should be in UserController

File: app/src/main/java/com/kms/cura/view/activity/LoginActivity.java [hunk 5]
```diff
@@ -188,4 +223,78 @@ protected void onResume() {
...
+            pbeCipher.init(Cipher.DECRYPT_MODE, key, new PBEParameterSpec(SALT, 20));
+            return new String(pbeCipher.doFinal(base64Decode(property)), "UTF-8");
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException | BadPaddingException | InvalidKeyException
+                | InvalidAlgorithmParameterException | NoSuchPaddingException | IOException | IllegalBlockSizeException e) {
+            e.printStackTrace();
+        }
+        return property;
+    }
+
+    private static String base64Encode(byte[] bytes) {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/LoginActivity.java]: This method should be in UserController

File: app/src/main/java/com/kms/cura/view/activity/LoginActivity.java [hunk 6]
```diff
@@ -188,4 +223,78 @@ protected void onResume() {
...
+            pbeCipher.init(Cipher.ENCRYPT_MODE, key, new PBEParameterSpec(SALT, 20));
+            return base64Encode(pbeCipher.doFinal(property.getBytes("UTF-8")));
+        } catch (NoSuchAlgorithmException | InvalidKeyException | NoSuchPaddingException | InvalidAlgorithmParameterException
+                | InvalidKeySpecException | BadPaddingException | IllegalBlockSizeException | UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+        return property;
+    }
+
+    private static String decrypt(String property) {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/LoginActivity.java]: This method should be in UserController

File: app/src/main/java/com/kms/cura/view/activity/LoginActivity.java [hunk 7]
```diff
@@ -188,4 +223,78 @@ protected void onResume() {
         registerEvent();
         super.onResume();
     }
+
+    private static String encrypt(String property) {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/LoginActivity.java]: This method should be in UserController

**Comment by toannguyen71195** [comment]
@hieumai Fixed

**Comment by hieumai** [comment]
OK from me. @vtnlinh95  and @huuduy216  please start verifying this.

**Comment by huuduy216** [comment]
verified ok

**Comment by vtnlinh95** [comment]
@hieumai verified ok

**Comment by hieumai** [comment]
@toannguyen71195  please resolve conflict and squash commits so that I can merge

**Pull request merged**